rules/windows/defense_evasion_masquerading_as_elastic_endpoint_process.toml

```
@@ -12,20 +12,25 @@ injection.
 """
 from = "now-9m"
 index = ["winlogbeat-*", "logs-endpoint.events.*"]
-language = "kuery"
+language = "eql"
 license = "Elastic License"
 name = "Suspicious Endpoint Security Parent Process"
 risk_score = 47
 rule_id = "b41a13c6-ba45-4bab-a534-df53d0cfed6a"
 severity = "medium"
 tags = ["Elastic", "Host", "Windows", "Threat Detection", "Defense Evasion"]
 timeline_id = "76e52245-7519-4251-91ab-262fb1a1728c"
-type = "query"
+type = "eql"
 
 query = '''
-event.category:process and event.type:(start or process_started) and
- process.name:(esensor.exe or "elastic-endpoint.exe" or "elastic-agent.exe") and
- not process.parent.executable:"C:\Windows\System32\services.exe"
+process where event.type in ("start", "process_started", "info") and
+ process.name in ("esensor.exe", "elastic-endpoint.exe", "elastic-agent.exe") and
+ /* add FPs here */
+  not wildcard(process.parent.executable,"*:\\Program Files*\\Elastic\\*",
+                                         "*:\\Windows\\System*\\services.exe",
+                                         "*:\\Windows\\System*\\WerFault*.exe") and
+  /* avoid null parent process executable */
+  wildcard(process.parent.executable, "*")
```

avoid leading `*` when possible. it's always more performant to add a drive letter. since our rules are easy for users to edit, they can always customize
```suggestion
  /* avoid null parent process executable */
  process.parent.executable != null and

  not wildcard(process.parent.executable,"C:\\Program Files\\Elastic\\*",
                                         "C:\\Windows\\System32\\services.exe",
                                         "C:\\Windows\\System32\\WerFault*.exe")
```

---

and switch the `wildcard(field, "*")` to `field != null`